add tabs

RomanK17 opened the pull request.

RomanK17 committed:
add tabs

andrey-kudinov commented on src/js/modules/tabs.js:
```diff
@@ -0,0 +1,42 @@
+const pageTabs = (headerSelector, tabsSelector, contentSelector, activeClass) => {
```
Старайся пж не использовать в функции больше 2-3 аргументов
Это делает её сложной в использовании
Особенно, если часть аргументов опциональные
Так же для стороннего человека будет не понятно для чего все эти аргументы
Если вложить объект как аргумент, станет всё значительно читабельнее

Материал по этому вопросу (2-3 минуты):
https://github.com/ryanmcdermott/clean-code-javascript#function-arguments-2-or-fewer-ideally

andrey-kudinov commented on src/js/modules/tabs.js:
```diff
@@ -0,0 +1,42 @@
+
+    header.addEventListener('click', (event) => {
+        const target = event.target;
+        if (target) {
+            if (target.classList.contains(tabsSelector.replace(/\./, '')) ||
```
В этом случае if можно объединить в один, чтобы избавиться от лишней сложности
Так же это выражение можно вынести в отдельную переменную для понятности кода
И можно использовать option chaining

andrey-kudinov reviewed (changes requested):
Мне очень понравился стиль твоего кода
Ты круто всё сделал 👍
Я написал несколько моментов как можно ещё улучшить

Так же я заметил что при навигации с клавиатуры я могу переключаться между табами, но не могу его открыть клавишей `enter`

Сделай пж чтобы открывалось не только по клику, но и по клавише `enter`

RomanK17 committed:
changes to comments

changes:
replace function creatingTabs arguments to object with properties;
create isValidTabsTarget for expression in if condition;
add optional chaining in isValidTabsTarget.

RomanK17 committed:
-

RomanK17 committed:
dlt 2 useless functions

RomanK17 committed:
add listener for key 'enter'
